torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -198,6 +187,43 @@ std::array<kir::Bool*, 2> ShiftPredicateInserter::getPredicate(
 
     shift_pred = makeAndExpr(
         shift_pred, ir_builder.ltExpr(shift_max_pred_idx, shift_max_offset));
+  }
+
+  return shift_pred;
+}
+
+kir::Bool* ShiftPredicateInserter::getPaddingPredicate(
```

Comment:
It's a little unfortunate that we need to call `getConsumerRootPredIndices` twice, but I don't think of any easy way to combine them.

Can you try reducing code duplication between `getShiftPredicate` and `getPaddingPredicate`? I'd merge them with a flag parameter indicating which one to do, then use it from the two functions.

Reply:
I merged them together in `getPredicate`.